environment/docker/power.go

```diff
@@ -296,11 +296,52 @@ func (e *Environment) Terminate(ctx context.Context, signal string) error {
 	// We set it to stopping then offline to prevent crash detection from being triggered.
 	e.SetState(environment.ProcessStoppingState)
 
+	// Send the initial signal to the container.
 	if err := e.client.ContainerKill(ctx, e.Id, signal); err != nil && !client.IsErrNotFound(err) {
 		return errors.WithStack(err)
 	}
 
-	e.SetState(environment.ProcessOfflineState)
+	// Wait for up to 10 seconds, polling every 500ms, to check if the container has stopped.
+	const checkInterval = 500 * time.Millisecond
+	const timeout = 10 * time.Second
```

Comment:
Could this be configurable? Some hosts may want to adjust this.

Reply:
Both or just the time interval?

As the timeout just matched containerStop